src/analysis/normed_space/basic.lean

```diff
@@ -1242,13 +1271,37 @@ variables {α}
 
 open normed_field
 
-/-- If there is a scalar `c` with `∥c∥>1`, then any element can be moved by scalar multiplication to
-any shell of width `∥c∥`. Also recap information on the norm of the rescaling element that shows
-up in applications. -/
-lemma rescale_to_shell {c : α} (hc : 1 < ∥c∥) {ε : ℝ} (εpos : 0 < ε) {x : E} (hx : x ≠ 0) :
-  ∃d:α, d ≠ 0 ∧ ∥d • x∥ < ε ∧ (ε/∥c∥ ≤ ∥d • x∥) ∧ (∥d∥⁻¹ ≤ ε⁻¹ * ∥c∥ * ∥x∥) :=
+/-- The product of two seminormed spaces is a seminormed space, with the sup norm. -/
+instance prod.semi_normed_space : semi_normed_space α (E × F) :=
+{ norm_smul_le := λ s x, le_of_eq $ by simp [prod.semi_norm_def, norm_smul, mul_max_of_nonneg],
+  -- TODO: without the next two lines Lean unfolds `≤` to `real.le`
+  add_smul := λ r x y, prod.ext (add_smul _ _ _) (add_smul _ _ _),
+  smul_add := λ r x y, prod.ext (smul_add _ _ _) (smul_add _ _ _),
```

Comment:
It looks like this is an old problem that has since been resolved by an underlying bug fix in Lean. mathlib seems to compile fine if you now remove these lines.